src/cipher.rs

```diff
@@ -0,0 +1,89 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_jubjub::JubJubAffine;
+use rand_core::{CryptoRng, RngCore};
+
+use aes_gcm::{
+    aead::{Aead, AeadCore, KeyInit},
+    Aes256Gcm, Key,
+};
+
+const NONCE_SIZE: usize = 12;
+
+/// Size of the extra encryption data required by the
+/// cipher: the nonce (12 bytes) and the tag (16 bytes)
+pub const EXTRA_DATA_SIZE: usize = NONCE_SIZE + 16;
+
+/// Encrypts a plaintext given a shared DH secret key, returning a vector
+/// containing a nonce and the ciphertext (which includes the tag)
+pub fn encrypt<R: RngCore + CryptoRng, const ENCRYPTION_SIZE: usize>(
+    secret_key: &JubJubAffine,
```

Comment:
Can we call it `shared_secret`? I always get confused when I read `secret_key` and then it's a jubjub-point

Reply:
Maybe... or maybe it could require the bytes instead of a point. Mmm..

Reply:
maybe just call it `secret` then? although in the end, isn't always going to be a shared secret? A point from the dhke? Why else would you want to encrypt something if not for someone else to decrypt?